test_phage_reference/nbconverted/0_create_reference_genomes.py

```diff
@@ -0,0 +1,252 @@
+# Bad char [0xC2] in string at byte 52
+# cds chromosome:3013928-3015481(+) name=fhpR ;locus_tag=PA2665;replicon_accession=NC_002516.2;product=Transcriptional activator of P. aeruginosa flavohemoglobin, FhpR
+# ```
+# 
+# The error seems to be associated with non-acii characters present. https://github.com/tseemann/abricate/issues/77
+# 
+# **Fix:** Manually changed `name=fhpR ;` to `name=fhpR;`
+
+# In[4]:
+
+
+os.makedirs(paths.BLAST_DIR, exist_ok=True)
+
+
+# In[5]:
+
+
+get_ipython().run_cell_magic('bash', '-s $paths.PAO1_REF $paths.PAO1_DB_DIR', '\nmakeblastdb -in $1 -dbtype nucl -parse_seqids -out $2')
+
+
+# In[6]:
+
+
+get_ipython().run_cell_magic('bash', '-s $paths.PA14_REF $paths.PA14_DB_DIR', '\nmakeblastdb -in $1 -dbtype nucl -parse_seqids -out $2')
+
+
+# ### BLAST phage sequences against PAO1/PA14 DB
+# 
+# http://www.metagenomics.wiki/tools/blast/blastn-output-format-6
+# 
+# | Column header | Description |
+# | --- | --- | 
+# | qseqid | query (e.g., unknown gene) sequence id |
+# | sseqid | subject (e.g., reference genome) sequence id |
+# | pident | percentage of identical matches |
+# | length | alignment length (sequence overlap) |
+# | mismatch | number of mismatches |
+# | gapopen | number of gap openings |
+# | qstart | start of alignment in query |
+# | qend | end of alignment in query |
+# | sstart | start of alignment in subject
+# | send | end of alignment in subject |
+# | evalue | expect value |
+# | bitscore | bit score|
+# 
+# E-value: expected number of chance alignments; the smaller the E-value, the better the match.
+
+# In[7]:
+
+
+get_ipython().run_cell_magic('bash', '-s $paths.PHAGE_REF $paths.PAO1_BLAST_RESULT $paths.PAO1_DB_DIR', 'blastn -query $1 -out $2 -db $3 -outfmt 6 ')
+
+
+# In[8]:
+
+
+get_ipython().run_cell_magic('bash', '-s $paths.PHAGE_REF $paths.PA14_BLAST_RESULT $paths.PA14_DB_DIR', 'blastn -query $1 -out $2 -db $3 -outfmt 6 ')
+
+
+# In[9]:
+
+
+pao1_blast_result = pd.read_csv(paths.PAO1_BLAST_RESULT, sep="\t", header=None)
+pa14_blast_result = pd.read_csv(paths.PA14_BLAST_RESULT, sep="\t", header=None)
+
+
+# In[10]:
+
+
+# Add column names described above
+col_names = [
+    "qseqid",
+    "sseqid",
+    "pident",
+    "length",
+    "mismatch",
+    "gapopen",
+    "qstart",
+    "qend",
+    "sstart",
+    "send",
+    "evalue",
+    "bitscore"
+    
+]
+
+
+# In[11]:
+
+
+# BLAST results for PAO1
+pao1_blast_result.columns = col_names
+print(pao1_blast_result.shape)
+print(pao1_blast_result["evalue"].max())
+pao1_blast_result.head()
+
+
+# In[12]:
+
+
+# BLAST results for PA14
+pa14_blast_result.columns = col_names
+print(pa14_blast_result.shape)
+print(pa14_blast_result["evalue"].max())
+pa14_blast_result.head()
+
+
+# ## Add non-duplicate phage sequences only
+# The smaller the E-value, the better the match. So we want to add phage sequences with high E-value (i.e. evalue > 0.05) or phage sequences that are not on this table at all, because there was no hit found.
```

Comment:
This is really helpful! I am also very rusty when it comes to BLAST and your explanation makes a lot of sense. I didn't play around with the threshold here. The threshold of 0.05 was based off of some documentation about BLAST and since there were many sequences that didn't have BLAST hits, I used those as the non-redundant sequences that I should include. That was my rough logic. But I agree with your points to be more careful about how I am selecting sequences from the BLAST results. Something I will come back to